cpp/src/Ice/SSL/OpenSSLTransceiverI.h

```diff
@@ -37,17 +37,18 @@ namespace Ice::SSL::OpenSSL
             const Ice::SSL::ClientAuthenticationOptions&);
 
         ~TransceiverI();
-        IceInternal::NativeInfoPtr getNativeInfo() final;
+        [[nodiscard]] IceInternal::NativeInfoPtr getNativeInfo() final;
 
-        IceInternal::SocketOperation initialize(IceInternal::Buffer&, IceInternal::Buffer&) final;
-        IceInternal::SocketOperation closing(bool, std::exception_ptr) final;
+        [[nodiscard]] IceInternal::SocketOperation initialize(IceInternal::Buffer&, IceInternal::Buffer&) final;
+        [[nodiscard]] IceInternal::SocketOperation closing(bool, std::exception_ptr) final;
         void close() final;
-        IceInternal::SocketOperation write(IceInternal::Buffer&) final;
-        IceInternal::SocketOperation read(IceInternal::Buffer&) final;
-        std::string protocol() const final;
-        std::string toString() const final;
-        std::string toDetailedString() const final;
-        Ice::ConnectionInfoPtr getInfo(bool incoming, std::string adapterName, std::string connectionId) const final;
+        [[nodiscard]] IceInternal::SocketOperation write(IceInternal::Buffer&) final;
+        [[nodiscard]] IceInternal::SocketOperation read(IceInternal::Buffer&) final;
```

Comment:
Why are these `nodiscard`? Inheritance?  Seems to be missing from `SecureTransportTranscieverI.h`. What's the pattern for overrides and adding `nodiscard`?

Reply:
See https://stackoverflow.com/questions/49576298/are-function-attributes-inherited

However, it appears that clang-tidy treats `[[nodiscard]]` as inherited.